test_annex_copy_to failed for me while building 0.11.1 for trusty (nd14.04) only

I didn't find any record of this failing before, so decided to document
```
======================================================================
FAIL: datalad.support.tests.test_annexrepo.test_annex_copy_to
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/build/datalad-0.11.1/.pybuild/pythonX.Y_2.7/build/datalad/tests/utils.py", line 772, in newfunc
    t(*(arg + (uri,)), **kw)
  File "/build/datalad-0.11.1/.pybuild/pythonX.Y_2.7/build/datalad/tests/utils.py", line 596, in newfunc
    return t(*(arg + (filename,)), **kw)
  File "/build/datalad-0.11.1/.pybuild/pythonX.Y_2.7/build/datalad/support/tests/test_annexrepo.py", line 1343, in test_annex_copy_to
    eq_(repo.copy_to(["copied2", "copied1", "existed"], "target"),
  File "/build/datalad-0.11.1/.pybuild/pythonX.Y_2.7/build/datalad/support/gitrepo.py", line 313, in newfunc
    result = func(self, files_new, *args, **kwargs)
  File "/build/datalad-0.11.1/.pybuild/pythonX.Y_2.7/build/datalad/support/annexrepo.py", line 3237, in copy_to
    files, ['--in', 'here', '--not', '--in', remote])
  File "/build/datalad-0.11.1/.pybuild/pythonX.Y_2.7/build/datalad/support/annexrepo.py", line 1411, in _get_expected_files
    'find', opts=expr, files=files
  File "/build/datalad-0.11.1/.pybuild/pythonX.Y_2.7/build/datalad/support/annexrepo.py", line 2410, in _run_annex_command_json
    **kwargs)
  File "/build/datalad-0.11.1/.pybuild/pythonX.Y_2.7/build/datalad/support/tests/test_annexrepo.py", line 1334, in ok_copy
    assert_in('copied1', kwargs['annex_options'])
AssertionError: 'copied1' not found in ['--json', '--in', 'here', '--not', '--in', 'target', '--', 'copied2']
```


edit 1: failed to replicate in local trusty chroot -- might be some fluke/race in our tests/annex

--- a/datalad/support/tests/test_annexrepo.py
+++ b/datalad/support/tests/test_annexrepo.py
@@ -1339,6 +1339,9 @@ def test_annex_copy_to(origin, clone):
 {"command":"copy","note":"to target ...", "success":true, "key":"akey2", "file":"copied2"}
 {"command":"copy","note":"checking target ...", "success":true, "key":"akey3", "file":"existed"}
 """, ""
+    # Note that we patch _run_annex_command, which is also invoked by _run_annex_command_json
+    # which is in turn invoked first by copy_to for "find" operation.
+    # TODO: provide a dedicated handling within above ok_copy for 'find' command
     with patch.object(repo, '_run_annex_command', ok_copy):
         eq_(repo.copy_to(["copied2", "copied1", "existed"], "target"),
             ["copied1", "copied2"])

--- a/datalad/tests/test_utils.py
+++ b/datalad/tests/test_utils.py
@@ -62,9 +62,17 @@ from ..utils import import_modules, import_module_from_file
 from ..utils import get_open_files
 from ..utils import map_items
 from ..utils import unlink
+from ..utils import CMD_MAX_ARG
 from ..support.annexrepo import AnnexRepo
 
-from nose.tools import ok_, eq_, assert_false, assert_equal, assert_true
+from nose.tools import (
+    assert_equal,
+    assert_false,
+    assert_greater,
+    assert_true,
+    eq_,
+    ok_,
+)
 from datalad.tests.utils import nok_, assert_re_in
 
 from .utils import with_tempfile, assert_in, with_tree
@@ -1172,3 +1180,10 @@ def test_map_items():
     c_mapped = map_items(add10, c)
     assert type(c) is type(c_mapped)
     eq_(c_mapped.items(), [(11,), (12, 13), (14, 15, 16)])
+
+
+def test_CMD_MAX_ARG():
+    # 100 is arbitrarily large small integer ;)
+    # if fails -- we are unlikely to be able to work on this system
+    # and something went really wrong!
+    assert_greater(CMD_MAX_ARG, 100)

--- a/datalad/utils.py
+++ b/datalad/utils.py
@@ -86,14 +86,17 @@ except:  # pragma: no cover
 # The last one would be the most conservative/Windows
 CMD_MAX_ARG_HARDCODED = 2097152 if on_linux else 262144 if on_osx else 32767
 try:
-    CMD_MAX_ARG = os.sysconf('SC_ARG_MAX') or CMD_MAX_ARG_HARDCODED
+    CMD_MAX_ARG = os.sysconf('SC_ARG_MAX')
+    assert CMD_MAX_ARG > 0
 except Exception as exc:
     # ATM (20181005) SC_ARG_MAX available only on POSIX systems
-    # so exception would be thrown e.g. on Windows.
+    # so exception would be thrown e.g. on Windows, or
+    # somehow during Debian build for nd14.04 it is coming up with -1:
+    # https://github.com/datalad/datalad/issues/3015
     CMD_MAX_ARG = CMD_MAX_ARG_HARDCODED
     lgr.debug(
-        "Failed to query SC_ARG_MAX sysconf, will use hardcoded value: %s",
-        exc)
+        "Failed to query or got useless SC_ARG_MAX sysconf, "
+        "will use hardcoded value: %s", exc)
 # Even with all careful computations we do, due to necessity to account for
 # environment and what not, we still could not figure out "exact" way to
 # estimate it, but it was shown that 300k safety margin on linux was sufficient.

